frontend/src/app/data-entry/shared/designators-control/designators-control.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, forwardRef, Input } from '@angular/core';
+import { ControlValueAccessor, FormControl, FormGroup, NG_VALUE_ACCESSOR, Validators } from '@angular/forms';
+import { actionIcons } from '@shared/icons';
+import { BehaviorSubject, Subject } from 'rxjs';
+import _ from 'underscore';
+
+@Component({
+    selector: 'lc-designators-control',
+    templateUrl: './designators-control.component.html',
+    styleUrls: ['./designators-control.component.scss'],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => DesignatorsControlComponent),
+            multi: true,
+        }
+    ],
+})
+export class DesignatorsControlComponent implements ControlValueAccessor {
+    @Input({ required: true }) formControl!: FormControl<string[]>;
+
+    designators$ = new BehaviorSubject<string[]>([]);
+    blur$ = new Subject<void>();
+    disabled = false;
+
+    addForm = new FormGroup({
+        designator: new FormControl<string>('', {
+            nonNullable: true,
+            validators: [
+                Validators.required,
+                Validators.minLength(1),
+            ]
+        })
+    });
+
+    actionIcons = actionIcons;
+
+    writeValue(obj: string[]): void {
+        this.designators$.next(obj);
+    }
+
+    registerOnChange(fn: (_: any) => void): void {
+        this.designators$.subscribe(fn);
```

Comment:
These subscriptions are not closed at the moment. For bonus points, please make sure they are canceled upon component destruction.